### PR TITLE
Expand RE_BREAKPOINT macro on ARM64

### PR DESCRIPTION
**Pull request by larsimmisch**
Works as expected on MacOS with clang

### CODE REVIEW

File: include/re_types.h [hunk 1]
```diff
@@ -81,6 +81,8 @@ typedef SSIZE_T ssize_t;
 /** Defines a soft breakpoint */
 #if (defined(__i386__) || defined(__x86_64__))
 #define RE_BREAKPOINT __asm__("int $0x03")
+#elif defined(__aarch64__)
+#define RE_BREAKPOINT __builtin_debugtrap()
```
Comment (sreimers) [inline include/re_types.h:85]: I think we can make this more generic with:

```c
#elif defined(__has_builtin) &&  __has_builtin(____builtin_debugtrap)
#define RE_BREAKPOINT __builtin_debugtrap()
```

Comment (larsimmisch) [reply include/re_types.h:85]: Yes, that's much better. I've changed it accordingly.

Comment (sreimers) [reply include/re_types.h:85]: Looks like we need a modification for MSVC
```c
#elif !defined(MSVC) && \
	defined(__has_builtin) && __has_builtin(__builtin_debugtrap)
```

Comment (larsimmisch) [reply include/re_types.h:86]: I'll have an eye on the validation run

Comment (larsimmisch) [reply include/re_types.h:86]: Looks good

**Comment by sreimers** [comment]
Thanks!

**Pull request merged**